packages/dds/merge-tree/src/mergeTree.ts

```diff
@@ -2226,11 +2227,11 @@ export class MergeTree {
                     { leaf: onLeaf, candidateSegment: newSegment, continuePredicate: continueFrom });
 
                 if (newSegment.parent === undefined) {
-                    throw new Error(`MergeTree insert failed: ${JSON.stringify({
-                        currentSeq: this.collabWindow.currentSeq,
-                        minSeq: this.collabWindow.minSeq,
-                        segSeq: newSegment.seq,
-                    })}`);
+                    throw new DataCorruptionError("MergeTree insert failed", {
+                        referenceSequenceNumber: this.collabWindow.currentSeq,
```

Comment:
this isn't referenceSequenceNumber, that's refSeq. currentSeq is the last seq seen by the merge tree

Reply:
would be good to have both

Reply:
CollaborationWindow here only has `currentSeq`, `minSeq`, and `localSeq` as far as I can tell.  Still not sure I'm parsing the nuances of "current" and "local".  Were you suggesting refSeq might come from somewhere else entirely?

Reply:
refSeq is a parameter to this function, so the refSeq of the op for this insert. collab window is the state of the merge tree itself, so current is the last seq that's ever been see by this object across all ops